src/core/static/wasm/bergamot-translator-worker.appendix.js

```diff
@@ -55,29 +53,17 @@ shortlist:
     - 50
 `;
 
-      console.log("modelConfig: ", modelConfig);
+    console.log("modelConfig: ", modelConfig);
 
-      // Instantiate the TranslationModel
-      model = new Module.TranslationModel(modelConfig);
-      modelFrom = from;
-      modelTo = to;
-      log(
-        `Model ${languagePair} loaded in ${(Date.now() - start) / 1000} secs`,
-      );
-    } else {
-      log(`Model ${languagePair} already loaded`);
-    }
+    // Instantiate the TranslationModel
+    model = new Module.TranslationModel(modelConfig);
+    const loadModelEnd = performance.now();
```

Comment:
Are you sure we want to measure it on the lowest(wasm) level and not closer to user experience? What if there is some extra code somewhere later and wrong timings will be reported? I guess we interested mostly in how long the user has to wait, right?

Reply:
Not sure, and ideally I'd like to include both high-level and low-level timings. Currently when there is only one type of timing in the ping, I chose the wasm-level measurements since that informs wasm optimization efforts the most. 

Both high-level and low-level measurements are shown in the extension console log after a translation:

**With model-loading:**
Translation of all text in tab with id 7 (426 words) took 11.745 secs (perceived as 36 words per second) across 2 translation engine requests (which took 0.872 seconds, operating at 489 words per second). Model loading took 9.088 seconds.

**Without model-loading:**
Translation of all text in tab with id 7 (426 words) took 1.008 secs (perceived as 423 words per second) across 2 translation engine requests (which took 0.902 seconds, operating at 472 words per second). Model loading took 0 seconds.

As seen above, high-level timings are affected by model loading, queuing and a bunch of possible browser-specific glitches and we need to think about which high-level timings are the most important to measure, eg is it something like "total time waiting for the currently visible parts of the website to be translated"? Let's discuss in a follow-up issue.

Reply:
> Let's discuss in a follow-up issue.

Filed issue for this: https://github.com/mozilla-extensions/bergamot-browser-extension/issues/94

Reply:
I agree that ideally there should be low-level and high-level metrics, but to not overcomplicate things we can interpret all current metrics as the highest-level metrics, the closest to user experience because it is the most important for decision making and issues investigation. It requires moving some of them up on abstractions level. We can leave it as is in this PR, up to you, but we should keep in mind that these timings might not correspond to what a user actually sees.

Reply:
Yes exactly. These are low-level metrics to aid in getting numbers on how well the engine performs across different hardware setups.

Reply:
My concern is that when we analyze it we don't see the real picture. It doesn't matter that model loading is fast on wasm level if some later code hangs and a user quits because of it. we should start with this high level timing as the most important for analyzing user behavior. At the same time I wouldn't introduce extra metrics for low level and high level timings for now, it's a lot of them already, so I suggest moving everything on the highest level and it will correspond more or less to low level performance if there are no bugs in UI but it will allow catching problems on UI level too if there are any. What do you think?

Reply:
If you insist, I'll change it, but we won't have much in the form of real users until we change metrics.yaml to include both kind of metrics, and figuring out and implementing the right high-level metric will take some time. Imo, the time it takes for all of a web page is a rather irrelevant user experience metric.

Reply:
I think "how long is my CPU fan going to make noise" is a relevant metric.

Reply:
@motin I don't insist, this is up to you. If you feel like it will take a lot of time, let's just make a note to rethink this in the future.

@kpu this is a good point, we didn't think of adding resource utilization metrics yet. I'll add it to our future plans list.